### PR TITLE
fix: reduce taosBenchmark memory usage

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

File: src/benchData.c [hunk 1]
```diff
@@ -981,6 +956,7 @@ int bindParamBatch(threadInfo *pThreadInfo, uint32_t batch, int64_t startTime) {
         }
         param->buffer_type = data_type;
         param->length = calloc(batch, sizeof(int32_t));
+        g_memoryUsage += batch * sizeof(int32_t);
```
Comment (sangshuduo) [inline src/benchData.c:959]: should check if calloc() failed

File: src/benchInsert.c [hunk 2]
```diff
@@ -60,6 +60,7 @@ static int getSuperTableFromServer(int db_index, int stb_index) {
         }
     }
     int32_t *tmp_length_list = calloc(columnIndex, sizeof(int32_t));
+    g_memoryUsage += columnIndex * sizeof(int32_t);
```
Comment (sangshuduo) [inline src/benchInsert.c:63]: should check if calloc() falied.

File: src/benchInsert.c [hunk 3]
```diff
@@ -85,9 +86,11 @@ static int getSuperTableFromServer(int db_index, int stb_index) {
     stbInfo->tagCount = tagIndex;
     tmfree(stbInfo->tags);
     stbInfo->tags = calloc(tagIndex, sizeof(Column));
+    g_memoryUsage += tagIndex * sizeof(Column);
```
Comment (sangshuduo) [inline src/benchInsert.c:89]: same here

File: src/benchInsert.c [hunk 4]
```diff
@@ -85,9 +86,11 @@ static int getSuperTableFromServer(int db_index, int stb_index) {
...
     tmfree(stbInfo->columns);
     stbInfo->columnCount = columnIndex;
     stbInfo->columns = calloc(columnIndex, sizeof(Column));
+    g_memoryUsage += columnIndex * sizeof(Column);
```
Comment (sangshuduo) [inline src/benchInsert.c:93]: here

File: src/benchInsert.c [hunk 5]
```diff
@@ -139,12 +142,14 @@ static int getSuperTableFromServer(int db_index, int stb_index) {
...
             stbInfo->columns[columnIndex].name = calloc(
                 1,
-                sizeof(strlen((char *)row[TSDB_DESCRIBE_METRIC_FIELD_INDEX])) +
+                strlen((char *)row[TSDB_DESCRIBE_METRIC_FIELD_INDEX]) +
                     1);
+            g_memoryUsage += strlen((char *)row[TSDB_DESCRIBE_METRIC_FIELD_INDEX]);
             stbInfo->columns[columnIndex].comment = calloc(
                 1,
-                sizeof(strlen((char *)row[TSDB_DESCRIBE_METRIC_FIELD_INDEX])) +
+                strlen((char *)row[TSDB_DESCRIBE_METRIC_FIELD_INDEX]) +
```
Comment (sangshuduo) [inline src/benchInsert.c:150]: here

File: src/benchInsert.c [hunk 6]
```diff
@@ -139,12 +142,14 @@ static int getSuperTableFromServer(int db_index, int stb_index) {
             stbInfo->columns[columnIndex].min = (RAND_MAX >> 1) * -1;
             stbInfo->columns[columnIndex].name = calloc(
                 1,
-                sizeof(strlen((char *)row[TSDB_DESCRIBE_METRIC_FIELD_INDEX])) +
+                strlen((char *)row[TSDB_DESCRIBE_METRIC_FIELD_INDEX]) +
```
Comment (sangshuduo) [inline src/benchInsert.c:145]: here

File: src/benchInsert.c [hunk 7]
```diff
@@ -1424,8 +1428,10 @@ static int startMultiThreadInsertData(int db_index, int stb_index) {
     uint64_t tableFrom = 0;
     uint64_t ntables = stbInfo->childTblCount;
     stbInfo->childTblName = calloc(stbInfo->childTblCount, sizeof(char *));
+    g_memoryUsage += stbInfo->childTblCount * 8;
```
Comment (sangshuduo) [inline src/benchInsert.c:1431]: here

File: src/benchInsert.c [hunk 8]
```diff
@@ -1424,8 +1428,10 @@ static int startMultiThreadInsertData(int db_index, int stb_index) {
...
     for (int64_t i = 0; i < stbInfo->childTblCount; ++i) {
         stbInfo->childTblName[i] = calloc(1, TSDB_TABLE_NAME_LEN);
+        g_memoryUsage += TSDB_TABLE_NAME_LEN;
```
Comment (sangshuduo) [inline src/benchInsert.c:1434]: here

File: src/benchInsert.c [hunk 9]
```diff
@@ -1530,6 +1538,7 @@ static int startMultiThreadInsertData(int db_index, int stb_index) {
                         stbInfo->lenOfCols * g_arguments->reqPerReq + 1024;
                 }
                 pThreadInfo->buffer = calloc(1, pThreadInfo->max_sql_len);
+                g_memoryUsage += pThreadInfo->max_sql_len;
```
Comment (sangshuduo) [inline src/benchInsert.c:1541]: max_sql_len is always MAX_SQL_LEN now?

**Review by sangshuduo — CHANGES_REQUESTED** [review]
memory allocation should check if success or failure for gracefully exit

File: src/benchInsert.c (hunk 9 above)
Comment (zhaoyanggh) [reply src/benchInsert.c:1541]: for restful and taosc insert, is always MAX_SQL_LEN, for stmt and schemaless, especially for influxdb line and telnet protocol, which need string buffer to insert, will not have the MAX_SQL_LEN limitation, instead it only depend on the batch size, so pThreadInfo->max_sql_len for those insertion is dynamic

**Review by afwerar — APPROVED** [review]
LGTM

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**